### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jaredramirez**

### CODE REVIEW

**Comment by pouncilt** [comment]
Be sure to give me time to code review your pull request.  If it is something you really need me to review immediately, send me text, phone call, and lastly email.

Thanks,

Tonté

**Comment by jaredramirez** [comment]
Okay, will do in the future. Was just trying to get it ready to publish quickly so I could use it in pmms-moblie.

Thanks,

Jared Ramirez
Web Consultant
(940) 368-7410
jared.ramirez@bytepushers.software


On June 28, 2016 at 5:58:13 PM, Pouncil Tonté (notifications@github.com<mailto:notifications@github.com>) wrote:

Be sure to give me time to code review your pull request. If it is something you really need me to review immediately, send me text, phone call, and lastly email.

Thanks,

Tonté

**Comment by pouncilt** [comment]
Understood.  :)

File: src/main/javascript/software.bytepushers.utils.DateUtility.js [hunk 1]
```diff
@@ -27,5 +27,28 @@
         }
         return 0;
     };
+    /*checks if string passed in is valid MM/DD/YYYY date*/
+    BytePushers.DateUtility.isDateString_MMDDYYYY = function (dateString) {
```
Comment (pouncilt) [inline src/main/javascript/software.bytepushers.utils.DateUtility.js]: Probably should rename method name to validateDateFormat and pass in a date format constant MMDDYY and then use a switch statement to delegate to various date formate validators to ensure the dateString is formatted correctly.  If we refactor it this way we can expand support more date format validation in the future.

File: src/main/javascript/software.bytepushers.utils.DateUtility.js [hunk 2]
```diff
@@ -27,5 +27,28 @@
...
+        /*credit: https://stackoverflow.com/questions/6177975/how-to-validate-date-with-format-mm-dd-yyyy-in-javascript/6178341#6178341*/
+
+        /* First check for the pattern */
```
Comment (pouncilt) [inline src/main/javascript/software.bytepushers.utils.DateUtility.js]: Is line 34 through 52 used somewhere else.  We might consider breaking down into smaller re-usable methods.

File: src/test/javascript/dateUtilitySpec.js [hunk 3]
```diff
@@ -8,5 +8,12 @@ define(['bytepushers', 'bytepushersDateUtility'], function(BytePushers) {
             var date1 = new Date(97,2,30), date2 = new Date(99, 6, 23);
             expect(BytePushers.DateUtility.date_sort_asc(date1, date2)).toEqual(-1);
         });
+
+        it("validate date string", function(){
```
Comment (pouncilt) [inline src/test/javascript/dateUtilitySpec.js]: Nice job on validating your logic works!  Keep up the good work.

File: src/test/javascript/genericPropertyFilterSpec.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+define(['bytepushers', 'bytepushersFiltersProptery'], function (BytePushers) {
+    describe("bytepushers date filter tests", function() {
```
Comment (pouncilt) [inline src/test/javascript/genericPropertyFilterSpec.js]: Nice job on validating your logic works!  Keep up the good work.

File: src/test/javascript/genericPropertyFilterSpec.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+      //began writing filter tests and really need to finish filters period
+
+        it("should define BytePushers.filters.GenericProptertyFilter", function() {
+            expect(BytePushers.filters.GenericProptertyFilter).toBeDefined();
+        });
+
+        it("should filter date based on property", function() {
```
Comment (pouncilt) [inline src/test/javascript/genericPropertyFilterSpec.js]: Nice job on validating your logic works!  Keep up the good work.

File: src/test/javascript/genericPropertyFilterSpec.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            var date1 = new Date(),
+                date2 = new Date('11/17/2000'),
+                date3 = new Date('05/20/1990'),
+                array = [{date: date1}, {date: date2}, {date: date3}];
+
+            expect(BytePushers.filters.GenericProptertyFilter.DatePropteryFilter(array, date3, 'date')[0].date).toEqual(date3);
+        });
+
+        it("should filter name based on property", function() {
```
Comment (pouncilt) [inline src/test/javascript/genericPropertyFilterSpec.js]: Nice job on validating your logic works!  Keep up the good work.

File: src/main/javascript/software.bytepushers.filters.GenericPropertyFilter.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+/*global BytePushers, console*/
+/*jslint unparam: true*/
+(function (BytePushers) {
+    'use strict';
+    BytePushers = BytePushers || {};
+    BytePushers.filters = BytePushers.namespace("software.bytepushers.filters");
+    BytePushers.filters.GenericProptertyFilter = BytePushers.namespace("software.bytepushers.filters.GenericProptertyFilter");
+
+    BytePushers.filters.GenericProptertyFilter.DatePropteryFilter = function (values, date, propertyName) {
```
Comment (pouncilt) [inline src/main/javascript/software.bytepushers.filters.GenericPropertyFilter.js]: We should probably put some jsDocs around this.

File: src/main/javascript/software.bytepushers.filters.GenericPropertyFilter.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        values.forEach(function (value, index, values) {
+            if (value[propertyName].valueOf() === date.valueOf()) {
+                filteredDates.push(value);
+            }
+        });
+
+        return filteredDates;
+    };
+
+    BytePushers.filters.GenericProptertyFilter.StringPropteryFilter = function (values, searchText, propertyName) {
```
Comment (pouncilt) [inline src/main/javascript/software.bytepushers.filters.GenericPropertyFilter.js]: We should probably put some jsDocs around this.

File: src/main/javascript/software.bytepushers.utils.converters.date.DateConverter.js [hunk 9]
```diff
@@ -78,6 +78,18 @@
     BytePushers.converters.DateConverter.convertToDate_YYYYMMDDThhmmsssTZD = function (iso8601DateString) {
         return BytePushers.converters.DateConverter.convertToISO8601Date(iso8601DateString);
     };
+    BytePushers.converters.DateConverter.convertToString_MMDDYYY = function (d, delimeter) {
```
Comment (pouncilt) [inline src/main/javascript/software.bytepushers.utils.converters.date.DateConverter.js]: We should probably put some jsDocs around this.

File: src/main/javascript/software.bytepushers.utils.converters.date.DateConverter.js [hunk 10]
```diff
@@ -147,6 +159,9 @@
     BytePushers.converters.DateConverter.convertToString = function (d, dateFormat, delimeter) {
         var date = null;
         switch (dateFormat) {
+        case BytePushers.converters.DateConverter.MMDDYYYY_DATE_FORMAT:
```
Comment (pouncilt) [inline src/main/javascript/software.bytepushers.utils.converters.date.DateConverter.js]: Nice!

File: src/main/javascript/software.bytepushers.utils.DateUtility.js (hunk 1 above)
Comment (pouncilt) [inline src/main/javascript/software.bytepushers.utils.DateUtility.js]: We should probably put some jsDocs around this.

**Pull request merged**